src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -665,6 +666,34 @@ export class EditDataGridPanel extends GridParentComponent {
 		}
 	}
 
+	private async revertCurrentCell(): Promise<void> {
+		this.cellRevertInProgress = true;
+		this.updateEnabledState(false);
+		this.dataService.revertCell(this.currentCell.row, this.currentCell.column - 1)
```

Comment:
Is there a reason you're not awaiting this? There's currently not reason for this function to be async (or even return a promise) - but I don't see why you don't just await these calls and get rid of the .then's in the first place (the caller can choose to not await if needed in certain places)

Reply:
It maybe worth addressing all the `then's` together in a separate PR though.. there seem to be a repetitive use in this class.. @smartguest can you open a new issue to address this feedback and do full cleanup to use async/await instead?

Reply:
Sure, as long as this works as expected. But the code here seems to expect that the refresh will actually be completed when the promise resolves, which currently isn't true

https://github.com/microsoft/azuredatastudio/pull/23981/files#diff-e068c591ac6cf01b71fff02ee11ac75e072e6ad3770bd95223906694112fabc2R741

So if you're going to immediately go and fix all the async/awaits then that seems fine, but if that's a "backlog cleanup" item then I'd definitely suggest making sure this is going to work as expected before merging in.

Reply:
Might as well do all async/awaits together in single (separate) commit.

Reply:
I should await this, this was an oversight